### PR TITLE
doc/rados: edit ceph-conf.rst (4 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/configuration/ceph-conf.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/configuration/ceph-conf.rst [hunk 1]
```diff
@@ -493,18 +493,12 @@ present in the monitors' configuration database, use ``ceph config dump``.
 Help
 ====
 
-You can get help for a particular option with:
+To get help for a particular option , run the following command:
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:496]: space before comma

File: doc/rados/configuration/ceph-conf.rst [hunk 2]
```diff
@@ -550,20 +544,29 @@ or:
...
+.. prompt:: bash $
+
+   ceph daemon <name> config help [option]
 
 Runtime Changes
 ===============
 
 In most cases, Ceph permits changes to the configuration of a daemon at
-runtime. This can be used for increasing or decreasing the amount of logging
+run time. This can be used for increasing or decreasing the amount of logging
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:564]: IMHO conventionally this is one word.

File: doc/rados/configuration/ceph-conf.rst [hunk 3]
```diff
@@ -550,20 +544,29 @@ or:
...
 output, for enabling or disabling debug settings, and for runtime optimization.
 
-Configuration options can be updated via the ``ceph config set`` command.  For
-example, to enable the debug log level on a specific OSD, run a command of this form:
+Use the ``ceph config set`` command to update configuration options. For
+example, to enable the debug log level on a specific OSD, run a command of the
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:568]: enable the most verbose debug

File: doc/rados/configuration/ceph-conf.rst [hunk 4]
```diff
@@ -572,14 +575,14 @@ example, to enable the debug log level on a specific OSD, run a command of this
 .. note:: If an option has been customized in a local configuration file, the
    `central config
    <https://ceph.io/en/news/blog/2018/new-mimic-centralized-configuration-management/>`_
-   setting will be ignored (it has a lower priority than the local
+   setting will be ignored (because it has a lower priority than the local
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:578]: Suggest losing the parenthetical

File: doc/rados/configuration/ceph-conf.rst [hunk 5]
```diff
@@ -572,14 +575,14 @@ example, to enable the debug log level on a specific OSD, run a command of this
...
    configuration file).
 
 Override values
 ---------------
 
-Options can be set temporarily by using the `tell` or `daemon` interfaces on
-the Ceph CLI. These *override* values are ephemeral, which means that they
+Options can be set temporarily by using the ``tell`` or ``daemon`` interfaces
+on the Ceph CLI. These *override* values are ephemeral, which means that they
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:585]: by using the Ceph CLI ``tell`` or ``daemon`` commands.

File: doc/rados/configuration/ceph-conf.rst [hunk 6]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
 
-will show you the (non-default) options for that daemon.  You can also look at a specific option with:
+To see a specific setting, run the following command:
 
 .. prompt:: bash $
 
    ceph config show osd.0 debug_osd
 
-or view all options (even those with default values) with:
+To see all settings (even those with default values), run the following
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:646]: s/even/including/

File: doc/rados/configuration/ceph-conf.rst [hunk 7]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
+command:
 
 .. prompt:: bash $
 
    ceph config show-with-defaults osd.0
 
-You can also observe settings for a running daemon by connecting to it from the local host via the admin socket.  For example:
+You can see all settings for a daemon that is currently running by connecting
+to it from the local host via the admin socket. For example, to dump all
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:654]: s/from/on/

File: doc/rados/configuration/ceph-conf.rst [hunk 8]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
 
 The Octopus release changed the way the configuration file is parsed.
 These changes are as follows:
 
-- Repeated configuration options are allowed, and no warnings will be printed.
-  The value of the last one is used, which means that the setting last in the file
-  is the one that takes effect. Before this change, we would print warning messages
-  when lines with duplicated options were encountered, like::
+- Repeated configuration options are allowed, and no warnings will be
+  displayed.  The value of the last option is used, which means that the
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:682]: s/The value of the last option is used, which//

displayed.  This means that the ..

File: doc/rados/configuration/ceph-conf.rst [hunk 9]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
+  setting that comes last in the file is the one that takes effect. Prior to
+  this change, Ceph displayed warning messages when lines containing duplicated
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:684]: duplicate

File: doc/rados/configuration/ceph-conf.rst [hunk 10]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
     warning line 42: 'foo' in section 'bar' redefined
-
-- Invalid UTF-8 options were ignored with warning messages. But since Octopus,
-  they are treated as fatal errors.
-
-- Backslash ``\`` is used as the line continuation marker to combine the next
-  line with current one. Before Octopus, it was required to follow a backslash with
-  a non-empty line. But in Octopus, an empty line following a backslash is now allowed.
-
+- Prior to Octopus, invalid UTF-8 options were ignored with warning messages.
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:688]: I think ", options containing invalid UTF-8 characters"

File: doc/rados/configuration/ceph-conf.rst [hunk 11]
```diff
@@ -620,84 +623,84 @@ Override values can be set in two ways:
...
-- Before Octopus, if no section name was specified in the configuration file,
-  all options would be set as though they were within the :confsec:`global` section. This is
-  now discouraged. Since Octopus, only a single option is allowed for
-  configuration files without a section name.
-
-.. |---|   unicode:: U+2014 .. EM DASH
-   :trim:
+- Prior to Octopus, if no section name was specified in the configuration file,
+  all options would be set as though they were within the :confsec:`global`
+  section. This approach is now discouraged. Since Octopus, any configuration
```
Comment (anthonyeleven) [inline doc/rados/configuration/ceph-conf.rst:703]: s/now //

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/configuration/ceph-conf.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:496]: Oh good Lord. Things like this just might strengthen your rhetorical case against vim.

File: doc/rados/configuration/ceph-conf.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:564]: I think probably I'm going to accept that and bow to convention, even though doing so deprives us of an unambiguous distinction between the adjectival "runtime" and the obviously substantive "run time".

File: doc/rados/configuration/ceph-conf.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:568]: I defer to your fresh eyes. Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 1 above)
Comment (anthonyeleven) [reply doc/rados/configuration/ceph-conf.rst:496]: bwah hah hah.  

>I walked in there and I said "Droids, I want to lose.  I mean, I want
to lose.  I want to see line editors on CRTs and nulls in my files.
Write 36 bit ascii that can't be read except with the monitor
filtering it.  I mean LOSE, LOSE, LOSE!"  And I started jumping up and
down yelling "LOSE, LOSE", and Mark Crispin came in and started
jumping up and down with me yelling "Hee, dakedo UNIX nanka wo
tsukatte, umaku ikanakute mo shiranai yo", and a DEC sales rep came
over, put an arm around my shoulder, and said "How'd you like me to
show you a *real* editor that has macros and things like that?  We
have one, it's called TV...."

File: doc/rados/configuration/ceph-conf.rst (hunk 2 above)
Comment (anthonyeleven) [reply doc/rados/configuration/ceph-conf.rst:564]: I recall past arguments over "filesystem" where I lost :(

File: doc/rados/configuration/ceph-conf.rst (hunk 3 above)
Comment (anthonyeleven) [reply doc/rados/configuration/ceph-conf.rst:568]: Context:  log levels range from 0 to 20, the original text implies there is only one.

File: doc/rados/configuration/ceph-conf.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:564]: I think this is a battle that goes on forever. I expect that sometime in 2024 or 2025 we will begin the process of looking at each of these cases individually.

File: doc/rados/configuration/ceph-conf.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:568]: I have added a note that communicates this context.

File: doc/rados/configuration/ceph-conf.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:578]: I've removed the parentheses.

File: doc/rados/configuration/ceph-conf.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:585]: Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 6 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:646]: Another good suggestion of the word "including". Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 7 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:654]: Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 8 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:682]: Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 9 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:684]: Fair enough. Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 10 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:688]: Accepted.

File: doc/rados/configuration/ceph-conf.rst (hunk 11 above)
Comment (zdover23) [reply doc/rados/configuration/ceph-conf.rst:703]: Good suggestion. Accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/52593 - Reef backport
https://github.com/ceph/ceph/pull/52594 - Quincy backport

**Pull request merged**